src/datepicker/datepicker.js

```diff
@@ -535,6 +536,30 @@ function($compile, $parse, $document, $rootScope, $position, dateFilter, datePar
         return (new Date(date1.getFullYear(), date1.getMonth(), date1.getDate()) - new Date(date2.getFullYear(), date2.getMonth(), date2.getDate()));
       };
 
+      if(timelessJsonMode) {
+        scope.jsonToDate = function(date) {
+          if (date && angular.isString(date) && date.indexOf('-') > -1) {
+            var update = date.indexOf('T') > -1;
+            date = date.substring(0, 10).split('-');
+            date = new Date(+date[0], +date[1] - 1, +date[2]);
+            if (update) {
+              ngModel.$setViewValue(dateFilter(date, dateFormat));
+              ngModel.$render(); 
+            }
+          }
+          return date;
+        };
+        
+        scope.$parent.jsonToTicks = function(date) {
```

Comment:
@wesleycho OK. I understand what's happening now.

The `datepickerJsonUtils` directive that I was using in the [original commit](https://github.com/davious/angular-ui-bootstrap/commit/145578efd7a9e4ce8278f6b249b948e8a98a2fff#diff-6240fc17e068eaeef7095937a1d63eaeR842) didn't have an isolated scope, so it was actually doing the same thing as this [latest commit](https://github.com/davious/angular-ui-bootstrap/commit/e30d284aeab91ed21bfd537837cf247032b6d9e5). The only difference is the earlier way required additional markup.

Now that I realize this, I'm personally OK with the function and property living on the parent scope.

An incidental thought I had is I could move the `oneDay` property into an optional function parameter, for example: `jsonToTicks(someJsonDate, -1)`